modules/utxo-lib/src/bitgo/UtxoTransaction.ts

```diff
@@ -8,29 +8,36 @@ export function varSliceSize(slice: Buffer): number {
   return varuint.encodingLength(length) + length;
 }
 
-export class UtxoTransaction extends bitcoinjs.Transaction {
+export class UtxoTransaction<TNumber extends number | bigint = number> extends bitcoinjs.Transaction<TNumber> {
   static SIGHASH_FORKID = 0x40;
   /** @deprecated use SIGHASH_FORKID */
   static SIGHASH_BITCOINCASHBIP143 = UtxoTransaction.SIGHASH_FORKID;
 
-  constructor(public network: Network, transaction: bitcoinjs.Transaction = new bitcoinjs.Transaction()) {
+  constructor(
+    public network: Network,
+    transaction: bitcoinjs.Transaction<TNumber> = new bitcoinjs.Transaction<TNumber>()
+  ) {
     super();
     this.version = transaction.version;
     this.locktime = transaction.locktime;
     this.ins = transaction.ins.map((v) => ({ ...v }));
     this.outs = transaction.outs.map((v) => ({ ...v }));
   }
 
-  static fromBuffer(
+  static fromBuffer<TNumber extends number | bigint = number>(
     buf: Buffer,
     noStrict: boolean,
+    amountType: 'number' | 'bigint' = 'number',
     network?: Network,
-    prevOutput?: bitcoinjs.TxOutput[]
-  ): UtxoTransaction {
+    prevOutput?: bitcoinjs.TxOutput<TNumber>[]
+  ): UtxoTransaction<TNumber> {
     if (!network) {
       throw new Error(`must provide network`);
     }
-    return new UtxoTransaction(network, bitcoinjs.Transaction.fromBuffer(buf, noStrict));
+    if (amountType !== 'number' && (getMainnet(network) === networks.dash || getMainnet(network) === networks.zcash)) {
+      throw new Error('dash and zcash must use number amount type; bigint amount type is recommended for doge only');
```

Comment:
What about the other coins? (btc, bch, ... )

Reply:
Strictly speaking, `bigint` _should_ work with the other coins (it just definitely doesn't work with zcash/dash).
Also, some tests use `bigint` with BTC so I only restricted dash and zcash to accommodate those tests.